src/main/kotlin/uk/gov/justice/digital/hmpps/visitallocationapi/model/entity/PrisonerDetails.kt

```diff
@@ -57,4 +57,42 @@ data class PrisonerDetails(
       )
   }
     .minus(this.negativeVisitOrders.count { it.type == VisitOrderType.PVO && it.status == NegativeVisitOrderStatus.USED })
+
+  fun deepCopy(): PrisonerDetails {
```

Comment:
could we achieve this with the kotlin copy() method? https://kotlinlang.org/docs/data-classes.html#copying